src/comps/core/Form.tsx

```diff
@@ -62,8 +64,16 @@ export const FlowForm = ({
     initialValues: {
       [nameKey]: initialValue,
     },
-    onSubmit: (values) => {
-      handleSubmit(values[nameKey]);
+    onSubmit: async (values) => {
+      const error = await handleSubmit(values[nameKey]);
+      try {
+        if (error && typeof error === "string") {
+          await formik.setFieldTouched(nameKey, true, true);
+          throw new Error(error);
+        }
+      } catch (e: any) {
+        formik.setErrors({ [nameKey]: e.message });
+      }
```

Comment:
qq: what is this doing?

Reply:
yeah that much was clear. Why do we use `setFieldTouched`? Why throwing and then catching the error right after, instead of calling directly `setErrors`?

Reply:
good point

Reply:
fixed